gradio/components.py

```diff
@@ -2761,6 +2765,139 @@ def style(self, *, full_width: Optional[bool] = None, **kwargs):
         return IOComponent.style(self, **kwargs)
 
 
+@document("click", "upload", "style")
+class UploadButton(Clickable, Uploadable, IOComponent, SimpleSerializable):
+    """
+    Used to create an upload button, when cicked allows a user to upload files that satisfy the specified file type or generic files (if file_type not set).
+    Preprocessing: passes the uploaded file as a {file-object} or {List[file-object]} depending on `file_count` (or a {bytes}/{List{bytes}} depending on `type`)
+    Postprocessing: expects function to return a {str} path to a file, or {List[str]} consisting of paths to files.
+    Examples-format: a {str} path to a local file that populates the component.
+    Demos:
+    """
+
+    def __init__(
+        self,
+        label: str = "Upload a File",
+        value: Optional[str | List[str] | Callable] = None,
+        *,
+        visible: bool = True,
+        elem_id: Optional[str] = None,
+        type: str = "file",
+        file_count: str = "single",
+        file_type: List[str] = ["file"],
```

Comment:
Tested and this works great! I like how you've made this parameter a list, but I think it's better to avoid using [lists as default arguments](https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/). I would set it to `None`, which allows any file to be uploaded. 

Secondly, I would rename this parameter to be `file_types` which in my mind makes it clearer that it expects a list.

Reply:
Thirdly, when I tried passing in a regular string instead of a list, the frontend crashed. It would better to raise an exception